pkg/combustion/kubernetes.go

```diff
@@ -88,45 +98,146 @@ func configureRKE2(ctx *image.Context) (string, error) {
 		return "", fmt.Errorf("copying RKE2 installer script: %w", err)
 	}
 
-	config, err := parseKubernetesConfig(ctx)
-	if err != nil {
-		return "", fmt.Errorf("parsing RKE2 config: %w", err)
+	if len(ctx.ImageDefinition.Kubernetes.Nodes) > 1 {
+		return configureMultiNodeRKE2(ctx)
 	}
 
-	cni, multusEnabled, err := extractCNI(config)
+	return configureSingleNodeRKE2(ctx)
+}
+
+func configureSingleNodeRKE2(ctx *image.Context) (string, error) {
+	zap.S().Info("Configuring single node RKE2 cluster")
+
+	serverConfig, err := parseKubernetesConfig(ctx, k8sServerConfigFile)
 	if err != nil {
-		return "", fmt.Errorf("extracting CNI from RKE2 config: %w", err)
+		return "", fmt.Errorf("parsing RKE2 server config: %w", err)
 	}
 
-	configFile, err := storeKubernetesConfig(ctx, config, image.KubernetesDistroRKE2)
-	if err != nil {
-		return "", fmt.Errorf("storing RKE2 config file: %w", err)
+	// Establish sane default values
+	setClusterCNI(serverConfig)
+	if ctx.ImageDefinition.Kubernetes.Network.APIVIP != "" {
+		appendClusterTLSSAN(serverConfig, ctx.ImageDefinition.Kubernetes.Network.APIVIP)
 	}
+	if ctx.ImageDefinition.Kubernetes.Network.APIHost != "" {
+		appendClusterTLSSAN(serverConfig, ctx.ImageDefinition.Kubernetes.Network.APIHost)
+	}
+	delete(serverConfig, serverKey)
 
-	installPath, imagesPath, err := ctx.KubernetesArtefactDownloader.DownloadArtefacts(
-		ctx.ImageDefinition.Image.Arch,
-		ctx.ImageDefinition.Kubernetes.Version,
-		cni,
-		multusEnabled,
-		ctx.CombustionDir,
-	)
+	if err = storeKubernetesConfig(ctx, serverConfig, k8sServerConfigFile); err != nil {
+		return "", fmt.Errorf("storing RKE2 server config file: %w", err)
+	}
+
+	installPath, imagesPath, err := downloadRKE2Artefacts(ctx, serverConfig)
 	if err != nil {
 		return "", fmt.Errorf("downloading RKE2 artefacts: %w", err)
 	}
 
+	var vipManifest string
+
+	if ctx.ImageDefinition.Kubernetes.Network.APIVIP == "" {
+		zap.S().Info("Virtual IP address for RKE2 cluster is not provided and will not be configured")
```

Comment:
```suggestion
		zap.S().Info("Virtual IP address for RKE2 cluster is not provided and will not be configured.")
```

Reply:
Punctuation is (usually) not something we use for log file statements.